file/delete_scheduler.cc

```diff
@@ -197,7 +196,7 @@ Status DeleteScheduler::MarkAsTrash(const std::string& file_path,
     cnt++;
   }
   if (s.ok()) {
-    sst_file_manager_->OnMoveFile(file_path, *trash_file);
+    s = sst_file_manager_->OnMoveFile(file_path, *trash_file);
```

Comment:
Similar to above - I'm a bit wary of changes that lead us to return non-ok after the main operation succeeds - mainly because I introduced a bug in this way (#7600).

When this function returns non-ok to `DeleteScheduler::DeleteFile()`, it attempts to immediately delete the original file. But if the non-ok originated in `OnMoveFile()`, the original file would have already been renamed successfully. I haven't checked what is the overall implication but it looks like `DeleteScheduler::DeleteFile()` would then fail with the file not found error. At least that is a bit confusing because the file wasn't expected to be found.

Reply:
Changed to PermitUncheckedError instead of return.